projects/media-viewer/src/lib/store/reducers/icp.reducer.ts

```diff
@@ -28,10 +29,22 @@ export function icpReducer (state = initialIcpSessionState,
         ...state,
         session,
         client: participantInfo.client,
-        presenter: participantInfo.presenter
+        presenter: participantInfo.presenter,
       };
     }
 
+    case fromIcpActions.ICP_PARTICIPANT_LIST_UPDATED: {
+      const updatedParticipants: any = action.payload;
+      const participants = [];
+      for (const id in updatedParticipants) {
```

Comment:
this could be done in a more functional way
const participants = Object.keys({...updatedParticipants}).map(id => ({ id: id, username: updatedParticipants[id] }))